pkg/registry/servicecatalog/instance/storage.go

```diff
@@ -147,6 +149,38 @@ func NewStorage(opts server.Options) (rest.Storage, rest.Storage, rest.Storage)
 		DeleteStrategy:          instanceRESTStrategies,
 		EnableGarbageCollection: true,
 
+		TableConvertor: tableconvertor.NewTableConvertor(
+			[]metav1beta1.TableColumnDefinition{
+				{Name: "Name", Type: "string", Format: "name"},
+				{Name: "Class", Type: "string"},
+				{Name: "Plan", Type: "string"},
+				{Name: "Status", Type: "string"},
+				{Name: "Age", Type: "string"},
+			},
+			func(obj runtime.Object, m metav1.Object, name, age string) ([]interface{}, error) {
+				getStatus := func(status servicecatalog.ServiceInstanceStatus) string {
+					if len(status.Conditions) > 0 {
+						condition := status.Conditions[len(status.Conditions)-1]
+						if condition.Status == servicecatalog.ConditionTrue {
+							return string(condition.Type)
+						}
+						return condition.Reason
+					}
+					return ""
+				}
+
+				instance := obj.(*servicecatalog.ServiceInstance)
+				cells := []interface{}{
+					name,
+					instance.Spec.GetSpecifiedServiceClass(),
```

Comment:
FYI, @eriknelson 's first pass on this resulted in two separate functions being created: one for cluster level classes/plans and another for namespaced. We spoke at the time about a follow-on PR that consolidated the two when being used by the svcat CLI so that we don't need awkward if/else statements because our helpers aren't helpful enough.

Just wanted you to be aware that this won't work the way you expect right now.

Reply:
Nevermind. I'll throw that function out, since I no longer need it. I'll let someone else deal with the pain of naming these two methods :)